### DISCUSSION TITLE
How can I hide empty vertical axis with column chart ?

### DISCUSSION THREAD

**Discussion by yoobi** [patrykandpatrick/vico#649] [Questions] [answered]
Hello ! 
I'm using version: `2.0.0-alpha.14`
I'm starting to use your great library but I can't find how to hide my empty vertical axis.
 Here is my code:
```kt
LaunchedEffect(deck) {
        modelProducer.tryRunTransaction {
            columnSeries {
                series(
                    x = listOf(1000, 3000, 4000, 5000, 6000, 7000, 9000),
                    y = listOf(4, 4, 8, 13, 4, 6, 4)
                )
            }
        }
    }
    CartesianChartHost(
        chart = rememberCartesianChart(
            rememberColumnCartesianLayer(),
            startAxis = rememberStartAxis(),
            bottomAxis = rememberBottomAxis(),
        ),
        modelProducer = modelProducer,
        modifier = modifier
    )
```

And it shows like below

<img width="339" alt="Screenshot 2024-04-08 at 11 16 07" src="https://github.com/patrykandpatrick/vico/assets/17434735/743fe9f6-8e60-43f3-9856-f1ec92fcca44">

**Comment by patrickmichalik** [comment]
Hello! Just to confirm, are you referring to the empty space on the right side of the chart?

**Reply by yoobi** [reply]
Hello, nope I was referring to the 2000 and 8000 x axis where there is no value, I've looked a bit more into the Vico API and I'm not sure if the solution to my issue is in the `getXStep` parameter

**Reply by patrickmichalik** [reply]
Got it, @yoobi. That was my initial suspicion, but then I noticed that there’s more empty space on the right side of the chart than on its left side. I see now that the extra space has a different color, so the asymmetry is, presumably, a matter of the screenshot’s cropping.

Please take a look [here](https://patrykandpatrick.com/vico/wiki/cartesian-charts/axes#category-axes). While you have numbers rather than letters, the same general approach can be used.

**Reply by patrickmichalik** [reply, marked as answer]
Here’s a quick example with your data, @yoobi:

```kt
val xKey = ExtraStore.Key<List<Int>>()

@Preview
@Composable
fun Example() {
    val modelProducer = remember { CartesianChartModelProducer.build() }
    LaunchedEffect(Unit) {
        modelProducer.tryRunTransaction {
            columnSeries { series(4, 4, 8, 13, 4, 6, 4) }
            updateExtras { it[xKey] = listOf(1000, 3000, 4000, 5000, 6000, 7000, 9000) }
        }
    }
    CartesianChartHost(
        rememberCartesianChart(
            rememberColumnCartesianLayer(),
            startAxis = rememberStartAxis(),
            bottomAxis =
                rememberBottomAxis(
                    valueFormatter =
                        remember {
                            { index, chartValues, _ -> "${chartValues.model.extraStore[xKey][index.toInt()]}" }
                        },
                ),
        ),
        modelProducer,
    )
}
```

<img src="https://github.com/patrykandpatrick/vico/assets/120058021/e285d5a2-e36e-47fb-8083-5edfd99bf1cb" width="720" />

**Reply by yoobi** [reply]
Yep thank you ! That's what I did without the `remember` I'll add it.
It seems that le list I'm passing to the `CartesianChartHost` is not in the right order, so I'll sort it before using it in `CartesianChartHost`

EDIT: why is your xKey out of the composable?

**Reply by patrickmichalik** [reply]
You’re welcome, @yoobi! Regarding `xKey` being at the top level, this primarily relates to what’s described in the warning box [here](https://patrykandpatrick.com/vico/wiki/cartesian-charts/data#extras). Based on your code snippet, it can’t be deduced where the `CartesianChartModelProducer` is stored, so a universal approach was appropriate for the example. More generally, there isn’t any particular benefit to putting the `ExtraStore.Key` in the composable. A top-level field keeps the composable simpler and has the advantage of not requiring the use of `remember`. That said, if your `CartesianChartModelProducer` is tied to the composable where `CartesianChartHost` is used, as in my example, then creating the `ExtraStore.Key` in the composable will produce the same result, so long as you use `remember`.
